docs/asyncapi.yaml

```diff
@@ -37,170 +37,113 @@ components:
                 maximum: 9999999999
                 example: 1234567890
               apiLink:
-                type: url
                 description: api link for calling file stream of statement file that has been published unto blob
                 example: 'https://myStatementRetreivalApiEndpoint/statement-receiver/statement/v1/FFC_PaymentStatement_SFI_2022_1234567890_2022090615023001.pdf'  
               documentType:
                 type: string
                 enum:
                   - Payment statement
                 description: document type
-                example: 'Payment statement'
+                example: Payment statement
               scheme:
                 type: string
                 enum:
                   - SFI
                 description: Scheme name
                 example: SFI
-          type: 'uk.gov.pay.statement.crm'
-          source: 'ffc-pay-statement-generator'
   schemas:
     PaymentStatement:
       type: object
       required:
       - businessName
-      - sbi
       - frn
+      - sbi
       - address
-      - scheme
-      - payments
       - funding
+      - fundingTotal
+      - payments
+      - scheme
       properties:
         businessName:
           $ref: '#/components/schemas/BusinessName'
-        sbi:
-          $ref: '#/components/schemas/SBI'
-        frn:
-          $ref: '#/components/schemas/FRN'
         email:
           $ref: '#/components/schemas/Email'
+        frn:
+          $ref: '#/components/schemas/FRN'
+        sbi:
+          $ref: '#/components/schemas/SBI'
         address:
           $ref: '#/components/schemas/Address'
-        scheme:
-          $ref: '#/components/schemas/Scheme'
-        payments:
+        funding:
           type: array
           items:
-            $ref: '#/components/schemas/Payment'
-        funding:
+            - $ref: '#/components/schemas/Funding'
+            - $ref: '#/components/schemas/FundingTotal'
```

Comment:
On Studio's sandbox, it states that there can be other items (which for this is true, but see if you can find an asyncapi format that represents this better, i.e. 1+ Fundings but only 1 FundingTotal)

![Screenshot 2023-02-10 at 11 43 07](https://user-images.githubusercontent.com/56440327/218084539-e5a1360e-570e-4922-84b6-12a4d3b33be5.png)

Here's an example of using `oneof`: https://github.com/asyncapi/generator/issues/427

Reply:
Also, if you could look at it stating that here can be other items for the message itself which shouldn't be the case as we should be validating against that

![Screenshot 2023-02-10 at 11 50 28](https://user-images.githubusercontent.com/56440327/218085743-161bfaa9-9f1f-4679-8741-feba238bff20.png)

Reply:
The best I can do in regards to the initial comment, with my current knowledge, is using a combination on anyOf and oneOf. I got the following information from searching around online:

oneOf – validates the value against exactly one of the subschemas
allOf – validates the value against all the subschemas
anyOf – validates the value against any (one or more) of the subschemas

Reply:
Though, in trying the above, it seems like it interferes with the example message which is not what we would want

Reply:
Nice, okay good research. Let's add these also to the Confluence page!